src/modules/SX1261.cpp

```diff
@@ -0,0 +1,46 @@
+#include "SX1261.h"
+
+// note that this is untested (lacking the hardware) and based purely on the datasheet.
+int16_t SX1261::setOutputPower(int8_t power) {  
+  // check allowed power range
+  if (!((power >= -17) && (power <= 14))) {
+    return(ERR_INVALID_OUTPUT_POWER);
+  }
+
+  // get current OCP configuration
+  uint8_t ocp = 0;
+  int16_t state = readRegister(SX126X_REG_OCP_CONFIGURATION, &ocp, 1);
+  if (state != ERR_NONE) {
+    return(state);
+  }
+
+  state = setOptimalLowPowerPaConfig(&power);
+
+  // set output power
+  // TODO power ramp time configuration
+  if (state == ERR_NONE) {
+    state = SX126x::setTxParams(power);
+  }
+
+  // restore OCP configuration
+  int16_t state2 = writeRegister(SX126X_REG_OCP_CONFIGURATION, &ocp, 1);
```

Comment:
The logic here is so that OCP is restored even if one of setPaConfig or setTxParams fail.

The state2 is used so that:
 - If either setOptimalLowPowerPaConfig or setTxParams fail, then that error message will be returned to the caller.
 - If both setOptimalLowPowerPaConfig and setTxParams pass, then the result of restoring OCP will be passed to the user.

Reply:
As per the previous comment - the only way for either of those to fail is SPI transfer error. If that occurs, it means there's most likely a problem with the hardware.

Also, when there are multiple state variables, we get into issues with priority - if all of them are an error state, which one should be reported?